src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -292,6 +312,68 @@ private static RouteHandlerFilterDelegate CreateFilterPipeline(MethodInfo method
         return filteredInvocation;
     }
 
+    private static Expression MapHandlerReturnTypeToValueTask(Expression methodCall, Type returnType)
+    {
+        if (returnType == typeof(void))
+        {
+            return Expression.Block(methodCall, Expression.Constant(new ValueTask<object?>(EmptyHttpResult.Instance)));
+        }
+        else if (returnType == typeof(Task))
+        {
+            return Expression.Call(ExecuteTaskWithEmptyResultMethod, methodCall);
+        }
+        else if (returnType == typeof(ValueTask))
+        {
+            return Expression.Call(ExecuteValueTaskWithEmptyResultMethod, methodCall);
+        }
+        else if (returnType.IsGenericType &&
```

Comment:
Should we special case `ValueTask<object?>` and `Task<object?>` so we don't unnecessarily wrap them?

Reply:
I was able to do this for `ValueTask<object?>` but the type checking on the `Expression.Condition` expects the arguments to strictly match so the `Task` has to be unwrapped and rewrapped as a `ValueTask`.